version: "3.5"

services:
  frontend:
    build:
      context: .
      dockerfile: ./docker/Dockerfile-frontend
    volumes:
      - ./front-end/:/app
    ports:
      - "3000:3000"

    command: npm run start-dev
    depends_on:
      - backend

  backend:
    build:
      context: .
      dockerfile: ./docker/Dockerfile-server
    volumes:
      - ./server/:/app
    ports:
      - "8000:8000"
    env_file:
      - ./server/.env
      - ../env/email.env
    command: python manage.py runserver 0.0.0.0:8000

    depends_on:
      - db

  db:
    image: "postgres:13.0"
    hostname: db
    environment:
      POSTGRES_PASSWORD: "db"
      POSTGRES_USER: "db"
      POSTGRES_DB: "db"
    ports:
      - "5432:5432"

  rabbitmq:
    hostname: rabbitmq
    image: rabbitmq:3.9.7
    environment:
      - RABBITMQ_DEFAULT_USER=admin
      - RABBITMQ_DEFAULT_PASS=mypass
    ports:
      - "5672:5672"
    restart: on-failure
    logging:
      driver: none

  celery:
    hostname: celery
    build:
      context: .
      dockerfile: ./docker/Dockerfile-server
    command: celery -A chess_app worker -l info
    volumes:
      - ./server/:/app
    depends_on:
      - db
      - rabbitmq
    env_file:
      - ./server/.env
      - ../env/email.env
